### PR TITLE
(MODULES-10432) Add mod_auth_openidc support

### PR DESCRIPTION
**Pull request by asieraguado**
Adds [mod_auth_openidc](https://github.com/zmartzone/mod_auth_openidc) and the parameters `auth_oidc`, `oidc_settings` to enable OpenID Connect authentication and authorization for a `vhost`.

### CODE REVIEW

**Comment by traylenator** [comment]
@asieraguado create a ticket in https://tickets.puppetlabs.com/projects/MODULES and link to this.

Generally this seems good to me.

**Comment by asieraguado** [comment]
Ticket created.
https://tickets.puppetlabs.com/browse/MODULES-10432

**Comment by sheenaajay** [comment]
Thank you @asieraguado for your contribution. The changes and unit test cases look good. It will be great if you can add some acceptance test to verify the changes.

**Comment by asieraguado** [comment]
Thank you @sheenaajay for your review. I included an acceptance test in `vhost_spec.rb` that verifies `auth_oidc` and its settings. There is one CI test failing for Ubuntu 14.04, but the same test should pass for Ubuntu 16.04 - the required package is available at least since this version [1]. 

[1] https://packages.ubuntu.com/xenial/web/libapache2-mod-auth-openidc

**Comment by traylenator** [comment]
@sheenaajay ubuntu 14.04 is EOL. Can it be removed as a test target?

**Comment by nilsjh** [comment]
Thanks, we're eagerly waiting for this fix to be merged. :+1:

**Comment by ekohl** [comment]
For now you can skip that os in the acceptance test. I'd also like to see this merged :)

File: spec/acceptance/vhost_spec.rb [hunk 1]
```diff
@@ -1197,4 +1197,37 @@ class { 'apache::mod::shib': }
       it { is_expected.to contain 'ShibCompatValidUser On' }
     end
   end
+
+  describe 'auth_oidc' do
```
Comment (michaeltlombardi) [inline spec/acceptance/vhost_spec.rb]:
```suggestion
  describe 'auth_oidc' unless: (os[:family] == 'ubuntu' && os[:release].to_f == '14.04') do
```

This should cause the test to skip against Ubuntu 14.04.

**Comment by asieraguado** [comment]
Thank you, applied the change to skip Ubuntu 14.04 in the test.

File: spec/acceptance/vhost_spec.rb [hunk 2]
```diff
@@ -1197,4 +1197,37 @@ class { 'apache::mod::shib': }
       it { is_expected.to contain 'ShibCompatValidUser On' }
     end
   end
+
+  describe 'auth_oidc', unless: (os[:family] == 'ubuntu' && os[:release].to_f == '14.04') do
```
Comment (michaeltlombardi) [inline spec/acceptance/vhost_spec.rb]:
```suggestion
  describe 'auth_oidc', unless: (os[:family] == 'ubuntu' && os[:release].to_f == 14.04) do
```

This was my fault for erroneously including quotes around the 14.04... floats are not strings...

Sorry!

Comment (asieraguado) [reply spec/acceptance/vhost_spec.rb]: I haven't noticed the mistake either, thanks!

**Pull request merged**